stylegan_code_finder/train_with_lightning.py

```diff
@@ -0,0 +1,151 @@
+import argparse
+import datetime
+import json
+import logging
+import os
+from pathlib import Path
+
+import torch
+import torch.distributed as dist
+import wandb
+from pytorch_training.distributed import synchronize
+
+import global_config
+import pathlib
+from training_builder.train_builder_selection import get_train_builder_class
+from lightning_modules.ligntning_module_selection import get_segmenter_class
+from utils.config import load_yaml_config, merge_config_and_args
+from utils.data_loading import get_data_loader
+from visualization.segmentation_plotter_lightning import SegmentationPlotter
+import pytorch_lightning as pl
+from pytorch_lightning.loggers import WandbLogger
+
+def sanity_check_config(config: dict):
+    if 'network' in config:
+        choices = ['DocUFCN', 'TransUNet', 'EMANet', 'PixelEnsemble']
+        assert config['network'] in choices, f'The network must be one of: {", ".join(choices)}'
+    if 'dataset' in config:
+        choices = ['wpi', 'dataset_gan']
+        assert config['dataset'] in choices, f'The dataset must be one of: {", ".join(choices)}'
+    with open(config['class_to_color_map']) as f:
+        class_to_color_map = json.load(f)
+    assert len(class_to_color_map) == config['num_classes'], f'The number of classes in the class_to_color_map must ' \
+                                                             f'be equal to the num_classes in the config'
+
+
+def setup_distributed(mpi_backend: str, rank: int, world_size: int):
+    os.environ['MASTER_ADDR'] = 'localhost'
+    os.environ['MASTER_PORT'] = '12355'
+    dist.init_process_group(mpi_backend, rank=rank, world_size=world_size)
+
+
+def cleanup():
+    dist.destroy_process_group()
```

Comment:
All this distributed stuff can be remove since Lightning will handle this. (Please double check that distributed training actually works)

Reply:
There's also more stuff below that can be removed (also in args)